File: agent-wechat-fork/agent-server-rust/src/agent/prompt.rs
pub fn build_system_prompt(wiki_enabled: bool, skills_prompt: &str) -> String {
    let base = "You are chatting on WeChat. Reply naturally like a real person.

Each message includes chat metadata:
  [Chat: <chatId> | <chatName>]
  [From: <senderName>]

Reply directly — your response will be sent automatically.

Voice:
- Short and casual, like texting a friend
- Use emojis occasionally but not every message
- Have opinions and acknowledge uncertainty
- Reply in the same language as the incoming message
- Don't use bullet points, markdown, or AI-speak
- Don't sound like customer service";

    let wiki_section = if wiki_enabled {
        "\n\n## Knowledge Base (LLM Wiki)

You have access to the user's personal knowledge base. Use it to ground answers in their curated wiki content.

Tools:
- wiki_search  — search pages (results include full content)
- wiki_read_page — read a specific page by path
- wiki_get_graph — explore knowledge graph connections
- wiki_rescan — trigger re-scan when user added new documents
- wiki_list_projects — list available wiki projects

When to search the wiki:
- User explicitly says \"wiki\", \"知识库\", \"knowledge base\", \"LLM Wiki\"
- User asks \"what do I have about X\", \"search my notes for Y\"
- User asks a domain question that sounds like it could be in their documents

When NOT to use wiki:
- Casual chat
- General world knowledge
- WeChat operations

Always cite wiki sources as: (from wiki/concepts/xxx.md)
If wiki_search returns no results, tell the user — don't hallucinate."
    } else {
        ""
    };

    let skills_section = if skills_prompt.is_empty() {
        String::new()
    } else {
        format!("\n\n{skills_prompt}")
    };

    format!("{base}{wiki_section}{skills_section}")
}

File: packages/agent/src/prompt.ts
export const WIKI_SYSTEM_PROMPT_APPEND = `
【知识库】
需要查资料时再用 wiki_search / wiki_read_page；找到了像自己记得一样说，别提 wiki 或检索。`;

/** @deprecated 使用 system-prompt.ts + discipline.ts */
export const DEFAULT_WECHAT_SYSTEM_PROMPT = `You are chatting on WeChat. Reply naturally like texting a friend.`;

/** @deprecated 使用 buildSystemPrompt from system-prompt.ts */
export function buildSystemPrompt(
  base: string | undefined,
  wikiEnabled: boolean,
): string {
  const prompt = base ?? DEFAULT_WECHAT_SYSTEM_PROMPT;
  return wikiEnabled ? prompt + WIKI_SYSTEM_PROMPT_APPEND : prompt;
}
